integration/spark/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonAddLoadCommand.scala

```diff
@@ -228,24 +228,17 @@ case class CarbonAddLoadCommand(
     model.setTableName(carbonTable.getTableName)
     val operationContext = new OperationContext
     operationContext.setProperty("isLoadOrCompaction", false)
-    val loadTablePreExecutionEvent: LoadTablePreExecutionEvent =
-      new LoadTablePreExecutionEvent(
-        carbonTable.getCarbonTableIdentifier,
-        model)
-    operationContext.setProperty("isOverwrite", false)
-    OperationListenerBus.getInstance.fireEvent(loadTablePreExecutionEvent, operationContext)
-    // Add pre event listener for index indexSchema
-    val tableIndexes = IndexStoreManager.getInstance().getAllCGAndFGIndexes(carbonTable)
-    val indexOperationContext = new OperationContext()
-    if (tableIndexes.size() > 0) {
-      val indexNames: mutable.Buffer[String] =
-        tableIndexes.asScala.map(index => index.getIndexSchema.getIndexName)
-      val buildIndexPreExecutionEvent: BuildIndexPreExecutionEvent =
-        BuildIndexPreExecutionEvent(
-          sparkSession, carbonTable.getAbsoluteTableIdentifier, indexNames)
-      OperationListenerBus.getInstance().fireEvent(buildIndexPreExecutionEvent,
-        indexOperationContext)
-    }
+    val (tableIndexes, indexOperationContext) = CommonLoadUtils.firePreLoadEvents(
```

Comment:
@VenuReddy2103 can you confirm if at all please the same function is being used, if not change there also.

Reply:
Done